python_modules/dagster/dagster/_core/definitions/resource_definition.py

```diff
@@ -265,6 +275,13 @@ def get_resource_requirements(
             yield ResourceDependencyRequirement(key=resource_key, source_key=source_key)
 
 
+def dagster_maintained_resource(
+    resource_def: ResourceDefinition,
+) -> Callable[[ResourceDefinition], ResourceDefinition]:
+    resource_def._dagster_maintained = True
```

Comment:
mutating the field here feels not ideal, but this is tricky since `resource_def` could be a user-specified subclass of `ResourceDefinition`, so no way we could reliably reconstruct e.g. a copy, so it's probably fine?